Readme file updates to main

Miramoop opened the pull request.

Commit:
Changes to Styles

Commit:
Switching CSS to SASS File

Commit:
Merge branch 'feature/addition-of-styling-fixes' into develop

Commit:
Updating basic structure of readme file

- Adding table of contents
- Updating the formatting

Commit:
Add logo to readme

Commit:
Add back to home link

Commit:
Add badges to the top of the readme

Commit:
Change table formatting

Commit:
Add made with badge

Commit:
Add Borders & Alignment

Commit:
Fix Border

Commit:
Alignment Fix

Commit:
Intro Section Table Formatting Fix

Commit:
Intro Section Photo & Title Formatting Fix

Commit:
Update to table of contents format

Commit:
Clone the Repo Border Fix

Commit:
Table Formatting Change

Commit:
Table Formatting Fix

Commit:
Introduction Formatting Fix

Commit:
Add original project link to intro

Commit:
Add icons to headers

Commit:
Update table of contents format

Commit:
Remove logos because they are causing errors

Commit:
Addition of Readme file images & Additional Steps Added

Commit:
Finish up adding instructions to the readme file

Commit:
Test new formatting for note and important sections

Commit:
Fixes to formatting for all note & important sections

Commit:
Add emojis to sections test

Commit:
Emojis added to each section header

Commit:
Added more reference links

Miramoop committed:
Merge pull request #3 from Miramoop/feature/readme-file-updates

Readme file updates to develop

misterhamm commented on src/components/QuizResults.js:
```diff
@@ -18,8 +19,8 @@ const QuizResults = ({
     return title
       .replace(/&/g, "&amp;")
-      .replace(/</g, "lt;")
-      .replace(/>/g, "gt;")
```
What is it that you are handling here? What was the issue that prompted this?

misterhamm commented on src/styles/headerStyles.scss:
```diff
@@ -0,0 +1,11 @@
+@import "variables", "mainStyles", "mediaQueries";;
```
Not sure we need to import all these? Maybe just the variables? I could be wrong, though

tstirtz-bzs replied:
Yeah, all the style files should be imported in a `index.scss` file typically.

misterhamm commented on src/styles/headerStyles.scss:
```diff
@@ -0,0 +1,11 @@
+    align-items: center;
+    color: $mainTextColor;
+    display: block;
+    margin: 0vh;
```
```suggestion
    margin: 0;
```

misterhamm commented on src/styles/quizCompleteStyles.scss:
```diff
@@ -0,0 +1 @@
+@import "variables", "mainStyles", "mediaQueries";
```
Looks like this isn't being used - get rid of clutter.

misterhamm commented on src/styles/_mediaQueries.scss:
```diff
@@ -0,0 +1,45 @@
+
```
Typically we wouldn't use a separate file for media queries. Best to keep all the styles for a given component in one place.

misterhamm commented on src/styles/_mainStyles.scss:
```diff
@@ -0,0 +1,60 @@
+  background-image: linear-gradient(180deg, slateblue, palegoldenrod);
+  display: flex;
+  flex-direction: column;
+  margin: 0vh;
```
```suggestion
  margin: 0;
```

misterhamm commented on src/styles/footerStyles.scss:
```diff
@@ -0,0 +1,12 @@
+  
+  footer a {
+    padding: 0.4vh;
+  }
```
```suggestion
footer {
    margin: 0;
    padding: 1vh;
    text-align: center;
    width: 100%;
    
    a {
        padding: 0.4vh;
    }
  }
```

misterhamm commented on src/styles/quizResultsStyles.scss:
```diff
@@ -0,0 +1,41 @@
+.video-container iframe {
+  width: 100%;
+  height: auto;
+}
```
```suggestion
.video-container {
  max-width: 560px;
  margin: auto;

    iframe {
      width: 100%;
      height: auto;
    }
}
```

misterhamm commented on src/styles/quizResultsStyles.scss:
```diff
@@ -0,0 +1,41 @@
+#spotifyLink:hover {
+  background-color: $hoverColor;
+  color: $mainTextColor;
+}
```
```suggestion
#spotifyLink {
  flex-grow: 1;
  transition-duration: 0.4s;
  
      &:hover {
          background-color: $hoverColor;
          color: $mainTextColor;
    }
}


```

misterhamm commented on src/styles/quizStyles.scss:
```diff
@@ -0,0 +1,24 @@
+    justify-content: center;
+  }
+  
+  .choices button {
```
Nest this selector like the other suggestions

tstirtz-bzs commented on src/styles/_mainStyles.scss:
Having styles in the name of the file is redundant since the file is in a `styles` directory.

Commit:
PR #4 Fixes

-Removal of unused function
-Fixed names of SASS files to make less redundant
-Organized SASS files with nesting
-Fixed margin format
-Put media queries in their respective SASS files
-Create an index.css that is compiled at runtime of all the scss merged together

Commit:
Added ability for sass changes to be watched

-Installed concurrently and added new commands into the package.json to allow for the local dev server to watch for any changes made to the sass and redeploy to the local server

Commit:
Removing unnecessary photo

tstirtz-bzs commented on src/styles/styles.css:
```diff
@@ -1,221 +1,17 @@
-/* To Do - Remove any unncessary CSS & Refactor */
```
What is the purpose of this `styles.css` files? It looks like this is base styles. Typically a `styles.css` is the name of the bundled css file. I would suggest changing this to `base.scss` or something along those lines, and reserve the `styles.css` name for the bundled styles file.

Commit:
Removing unnecessary styles file & reworking the bundling of styles

Commit:
Renaming _styles.scss to base.scss
